RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -1414,8 +1593,7 @@ boolean acquireHP(){
  * returns true if my_hp() >= goal after attempting to restore.
  */
 boolean acquireHP(int goal){
-  int meat_reserve = can_interact() ? 0 : my_meat();
```

Comment:
I suspect the rationale for this was that healing should not cost meat as we want the user to have Cocoon perm'd as a minimum but then failed to account for paths where you don't have a healing skill and need to use items (e.g. AoWoL).

Reply:
Yeah, does this still properly heavily skew towards cocoon or other healing skills when available?

Reply:
>I suspect the rationale for this was that healing should not cost meat as we want the user to have Cocoon perm'd as a minimum but then failed to account for paths where you don't have a healing skill and need to use items (e.g. AoWoL).

That sounds right to me. Good intent but implementation is too harsh.

Also step 4 of restore optimization is sorting our restorers based on meat cost and if I understand it right removing all those who do not tie for cheapest. So either it will leave us with only 1 restorer costing the least meat, or we get multiple restorers tying at 0 meat which will then be further sorted in subsequent steps.
Should I try to analyze some debug logs to see if step 4 is working correctly?

Reply:
Ok, it seems to work fine:
```
> [DEBUG] - Rank 1 optimization, prefer to... remove negative status effects
> [DEBUG] - Rank 2 optimization, prefer to... maintain soft reserve limit (keep at least N on hand if possible)
> [DEBUG] - Removed from consideration: (a relaxing hot tub, hp: 243.0, mp: 0.0, negative effects remaining: 0.0)
> [DEBUG] - Rank 3 optimization, prefer to... try not to spend coinmaster tokens, maximizing hp/mp restored per token spent if we must spend
> [DEBUG] - Rank 4 optimization, prefer to... try not to spend meat, maximizing hp/mp restored per meat spent if we must spend
> [DEBUG] - Removed from consideration: (doc galaktik's homeopathic elixir, hp: 243.0, mp: 0.0, negative effects remaining: 0.0)
> [DEBUG] - Removed from consideration: (disco nap, hp: 243.0, mp: 0.0, negative effects remaining: 0.0)
> [DEBUG] - Removed from consideration: (doc galaktik's pungent unguent, hp: 243.0, mp: 0.0, negative effects remaining: 0.0)
> [DEBUG] - Removed from consideration: (lasagna bandages, hp: 243.0, mp: 0.0, negative effects remaining: 0.0)
> [DEBUG] - Removed from consideration: (tongue of the walrus, hp: 243.0, mp: 0.0, negative effects remaining: 0.0)
> [DEBUG] - Rank 5 optimization, prefer to... try to spend less mp, maximizing hp restored per mp spent if we must spend
> [DEBUG] - Rank 6 optimization, prefer to... minimize hp/mp shortage to goal and wasted hp/mp over max
> [DEBUG] - Removed from consideration: (cannelloni cocoon, hp: 543.0, mp: 0.0, negative effects remaining: 0.0)
> [DEBUG] - Rank 7 optimization, prefer to... minimize number of uses needed to reach goal

cast 2 Blood Bond
You acquire an effect: Blood Bond (20)

cast 8 Blood Bubble
You acquire an effect: Blood Bubble (24)
> [INFO] - Using item scroll of drastic healing as restore.

use 1 scroll of drastic healing
```
reason it went for scroll is because items already in inventory are considered to be free while in ronin/hardcore (prior to my 4th commit here they were considered to be free always. the code meant it to be only in ronin/hardcore but due to an error in implementation which I fixed it was always considering them to be free.)

Reply:
A thought occurs. Even if we are sticking with not considering mall prices of things already in inventory while in run, we should at least consider the autosell value. For example, the scroll of drastic healing could be sold for 107 meat, which makes using it less efficient than using cocoon unless the mp to cast cocoon would cost more than 107 meat to replenish. I haven't actually checked if this is the case, but you were saying things in inventory were considered free when in run, which is definitely bad if so.

Reply:
that is a good idea.